test/test.py

```diff
@@ -47,6 +47,17 @@ def test_hello_world_mwe(self):
 
         self.assertEqual(output, 'Hello!')
 
+    def test_build_types(self):
+        for build_type in ['release', 'relwithdebinfo', 'debug', 'coverage']:
```

Comment:
I think a better test would be to get the compile command for a trivial example and check that the command is what would be expected. Here we only check that nothing crashes. Maybe this becomes easier, once we think a bit more about a good API, where things like getting the final build command out should be easy.

Reply:
That is a good idea!
Previously, the 'coverage' build type did not actually work at all, this test at least ensures it's theoretically usable.